File: backend/app/schemas/audit.py
from pydantic import BaseModel, ConfigDict
from typing import Optional
from datetime import datetime

class AuditBase(BaseModel):
    event_id: str
    user_id: str
    user_name: str
    role: str
    case_id: Optional[str] = None
    document_id: Optional[str] = None
    action: str
    purpose: Optional[str] = None
    result: str
    risk_level: str
    description: str
    ip_address: Optional[str] = "10.240.12.84"

class AuditCreate(BaseModel):
    case_id: Optional[str] = None
    document_id: Optional[str] = None
    action: str
    purpose: Optional[str] = None
    result: str
    risk_level: str
    description: str

class AuditResponse(AuditBase):
    timestamp: Optional[datetime] = None

    model_config = ConfigDict(from_attributes=True)

File: Frontend/src/types/audit.ts
import { Role } from './auth';

export type AuditEventType =
  | 'LOGIN'
  | 'LOGOUT'
  | 'CASE_CREATED'
  | 'CASE_UPDATED'
  | 'CASE_VIEW'
  | 'DOCUMENT_VIEW'
  | 'DOCUMENT_DOWNLOAD'
  | 'DOCUMENT_SHARE'
  | 'ACCESS_GRANTED'
  | 'ACCESS_DENIED'
  | 'DOCUMENT_UPLOAD'
  | 'DOCUMENT_VERSION_CREATED'
  | 'INTEGRITY_VERIFICATION'
  | 'CASE_ASSIGNMENT_CHANGED'
  | 'SECURITY_ALERT';

export type AccessPurpose =
  | 'INVESTIGATION'
  | 'FORENSIC_ANALYSIS'
  | 'LEGAL_REVIEW'
  | 'COURT_PROCEEDING'
  | 'AUDIT'
  | 'ADMINISTRATION';

export interface AuditEntry {
  eventId: string;
  timestamp: string;
  userId: string;
  userName: string;
  role: Role;
  caseId?: string;
  documentId?: string;
  action: AuditEventType;
  purpose?: AccessPurpose | string;
  result: 'SUCCESS' | 'DENIED' | 'FLAGGED';
  riskLevel: 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';
  description: string;
  ipAddress?: string;
}

export interface AuditSummary {
  totalRecentEvents: number;
  successfulAccesses: number;
  deniedAttempts: number;
  evidenceVerificationEvents: number;
  highRiskEvents: number;
}

